### PR TITLE
[BEAM-2799] list supported SQL grammar in BNF

### PR DESCRIPTION
**Pull request by mingmxu**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/649/

Jenkins built the site at commit id 1a202a6cc19769079ed07569e11b3313b96fbc33 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/300/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by XuMingmin** [comment]
R: @takidau @xumingming

File: src/documentation/dsls/sql.md [hunk 1]
```diff
@@ -112,18 +112,65 @@ Both methods wrap the back-end details of parsing/validation/assembling, and del
...
-The following features are supported in current repository:
+```
+query:
+	{
+          select
+      |   query UNION [ ALL ] query
+      |   query MINUS [ ALL ] query
+      |   query INTERSECT [ ALL ] query
+	}
+    [ ORDER BY orderItem [, orderItem ]* LIMIT [offset] count ]
```
Comment (takidau) [inline src/documentation/dsls/sql.md]: Do we really have ORDER BY support? Can you point me to the unit tests if so?

Comment (XuMingmin) [reply src/documentation/dsls/sql.md]: Yes, it's supported, but `ORDER BY` should always follow with a `LIMIT`.

Here's the [code](https://github.com/apache/beam/blob/DSL_SQL/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java) to support `ORDER BY`, and test cases [here](https://github.com/apache/beam/blob/DSL_SQL/sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRelTest.java#L75-L78)

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/654/

Jenkins built the site at commit id 10c8142bf3e9cbc3982ca12ed65e5d77bf6516fe with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/300/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

File: src/documentation/dsls/sql.md (hunk 1 above)
Comment (takidau) [reply src/documentation/dsls/sql.md]: Excellent, thank you!

**Comment by XuMingmin** [comment]
thank you @takidau . 
Btw, merge-bot is awesome lol

**Comment by XuMingmin** [comment]
@asfgit merge

**Comment by asfgit** [comment]
MergeBot: Merge: PR 300 in state error with description Unexpected Error: 502 Server Error: Proxy Error for url: https://builds.apache.org/job/beam_PreCommit_Website_Merge/api/python?tree=builds%5Bnumber%2Curl%5D.. URL: http://mergebot-vm.apache.org/beam-website/300.

**Comment by asfgit** [comment]
Error: MergeBot encountered an unexpected error while processing this PR: 502 Server Error: Proxy Error for url: https://builds.apache.org/job/beam_PreCommit_Website_Merge/api/python?tree=builds%5Bnumber%2Curl%5D.

**Comment by XuMingmin** [comment]
@asfgit merge

**Comment by asfgit** [comment]
Error: MergeBot encountered an unexpected error while processing this PR: 502 Server Error: Proxy Error for url: https://builds.apache.org/api/python?tree=jobs%5Bname%2Ccolor%2Curl%5D.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/655/

Jenkins built the site at commit id 3e8980e4ec3f84dd38397f0d28aa52ea3bf12a07 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/300/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by XuMingmin** [comment]
@asfgit merge

**Comment by XuMingmin** [comment]
closed #300

**Pull request merged**